pkg/actions/lua/databricks/client.go

```diff
@@ -0,0 +1,177 @@
+		return "", errors.Join(ErrTableCreationFailure, err)
+	}
+	statement := fmt.Sprintf(`CREATE EXTERNAL TABLE %s LOCATION '%s'`, tableName, location)
+	esr, err := client.workspaceClient.StatementExecution.ExecuteAndWait(client.ctx, sql.ExecuteStatementRequest{
+		WarehouseId: warehouseID,
+		Catalog:     catalogName,
+		Schema:      schemaName,
+		Statement:   statement,
+	})
+	if err != nil {
+		return "", fmt.Errorf("external table creation failed: %w", err)
+	}
+	return esr.Status.State.String(), nil
+}
+
+func tableFullName(catalogName, schemaName, tableName string) string {
+	return fmt.Sprintf("%s.%s.%s", catalogName, schemaName, tableName)
+}
+
+func (client *Client) deleteTable(catalogName, schemaName, tableName string) error {
+	err := client.workspaceClient.Tables.DeleteByFullName(client.ctx, tableFullName(catalogName, schemaName, tableName))
+	if err != nil {
+		return fmt.Errorf("failed deleting an existing table: %w", err)
+	}
+	return nil
+}
+
+func (client *Client) createOrGetSchema(catalogName, schemaName string) (*catalog.SchemaInfo, error) {
+	schemaInfo, err := client.workspaceClient.Schemas.Create(client.ctx, catalog.CreateSchema{
+		Name:        schemaName,
+		CatalogName: catalogName,
+	})
+
+	if err == nil {
+		return schemaInfo, nil
+	}
+	if alreadyExists(err) {
+		// Full name of schema, in form of <catalog_name>.<schema_name>
+		schemaInfo, err = client.workspaceClient.Schemas.GetByFullName(client.ctx, catalogName+"."+schemaName)
+		if err == nil {
+			return schemaInfo, nil
+		}
+	}
+	return nil, err
+}
+
+func newDatabricksClient(l *lua.State) (*databricks.WorkspaceClient, error) {
+	host := lua.CheckString(l, -2)
+	token := lua.CheckString(l, -1)
+	return databricks.NewWorkspaceClient(
+		&databricks.Config{
+			Host:        host,
+			Token:       token,
+			Credentials: config.PatCredentials{},
+		},
+	)
+}
+
+func (client *Client) RegisterExternalTable(l *lua.State) int {
+	tableName := lua.CheckString(l, 1)
+	location := lua.CheckString(l, 2)
+	warehouseID := lua.CheckString(l, 3)
+	catalogName := lua.CheckString(l, 4)
+	schemaName := lua.CheckString(l, 5)
+
+	status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location)
+	if err != nil {
+		if alreadyExists(err) {
+			err = client.deleteTable(catalogName, schemaName, tableName)
+			if err != nil {
+				lua.Errorf(l, err.Error())
+				panic("unreachable")
+			}
+			status, err = client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location)
+			if err != nil {
+				lua.Errorf(l, err.Error())
+				panic("unreachable")
+			}
+		} else {
+			lua.Errorf(l, err.Error())
+			panic("unreachable")
+		}
+	}
+	l.PushString(status)
+	return 1
+}
+
+func (client *Client) CreateOrGetSchema(l *lua.State) int {
+	ref := lua.CheckString(l, 1)
+	catalogName := lua.CheckString(l, 2)
+	schemaInfo, err := client.createOrGetSchema(catalogName, ref)
+	if err != nil {
+		lua.Errorf(l, err.Error())
```

Comment:
Not related to this PR - can you open an error to fix the usage of `lua.Errorf`.
It accepts state, format and args, like Printf.
So in this case we should call `lua.Errorf(l, "%s", err.Error())` unless you 100% sure there are no fmt patterns inside the error.

Reply:
https://github.com/treeverse/lakeFS/issues/7185